### DISCUSSION TITLE
access a field with operator[]

### DISCUSSION THREAD

**Discussion by sarikh** [nlohmann/json#2822] [Q&A] [answered]
assuming I have a field called "aaa" that does not always exist in my json, is there a different way to access it other than the following:
 `int a;
if (req_json.contains("aaa"))
        {
            a= req_json["aaa"];
        }
        else
        {
            a= 0;
        }`

**Comment by nlohmann** [comment, marked as answer]
For this use case, you can use the `value` function (https://json.nlohmann.me/api/basic_json/value/):

```cpp
int a = req_json.value("aaa", 0);
```
